### PR TITLE
Fix missing null check on `AbstractQuery#setResultCacheProfile()`

### PR DESCRIPTION
**Pull request by KonstantinKuklin**
subject

### CODE REVIEW

**Comment by Ocramius** [comment]
While the failure is trivial, we still need a test case for this

**Comment by KonstantinKuklin** [comment]
@Ocramius done

File: tests/Doctrine/Tests/ORM/QueryTest.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+    /**
+     * setResultCacheProfile should works correct with null
+     */
+    public function testSetResultCacheProfile()
+    {
+        $query = $this->_em->createQuery('SELECT u FROM ' . CmsUser::class . ' u');
+        $result = $query->setResultCacheProfile(null);
+        self::assertEquals($result, $query);
```
Comment (jwage) [inline tests/Doctrine/Tests/ORM/QueryTest.php]: I think we should be using `$this` instead of `self::` like the rest of the code base right?

Comment (alcaeus) [reply tests/Doctrine/Tests/ORM/QueryTest.php]: Strictly speaking, the `assert*` methods in PHPUnit are static, so they should be called using `self::assert*`, not `$this->assert*`. IIRC @Ocramius and @lcobucci have started changing these in new tests and for v3.

**Comment by KonstantinKuklin** [comment]
so, any updates here guys?
@Ocramius , @jwage

File: tests/Doctrine/Tests/ORM/QueryTest.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    protected function setUp()
+    {
+        $this->_em = $this->_getTestEntityManager();
+    }
+
+    /**
+     * setResultCacheProfile should works correct with null
+     */
+    public function testSetResultCacheProfile()
```
Comment (lcobucci) [inline tests/Doctrine/Tests/ORM/QueryTest.php]: Shouldn't this be part of https://github.com/doctrine/doctrine2/blob/f2dc9a8f92b7eb9a566b3ba7af16559cc2bc0eb7/tests/Doctrine/Tests/ORM/Query/QueryTest.php?

File: tests/Doctrine/Tests/ORM/QueryTest.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    {
+        $query = $this->_em->createQuery('SELECT u FROM ' . CmsUser::class . ' u');
+        $result = $query->setResultCacheProfile(null);
```
Comment (lcobucci) [inline tests/Doctrine/Tests/ORM/QueryTest.php]: This is a bit misleading, `$result` is not the query result but rather the query

File: tests/Doctrine/Tests/ORM/QueryTest.php (hunk 1 above)
Comment (lcobucci) [reply tests/Doctrine/Tests/ORM/QueryTest.php]: Shouldn't we assert that the cache profile was NOT configured?

**Review by lcobucci — CHANGES_REQUESTED** [review]

**Comment by Majkl578** [comment]
Linking #6861 fixing the same bug, but missing tests.

**Review by lcobucci — APPROVED** [review]

**Comment by lcobucci** [comment]
@KonstantinKuklin 🚢 and backported via 56ebf4bea91d4c0bbcf411c43f2f322a1a65fafa, thanks for your contribution!

**Pull request merged**